community/planning/publisher_for_scabbard_v3.md

```diff
@@ -54,144 +54,150 @@ continue to be added and scheduled if they are available.
-}
+/// The context used by the publisher
+pub trait PublishingContext: Send {}
 ```
 
-#### Batch and Transaction
+#### Batch
 
-The trait for a batch that will be used by the publisher. The transactions in a
-batch must also be defined by a trait. These trait will live in sawtooth-lib and
-is used to define one of the generics for the publisher.
+The trait for a batch that will be used by the publisher. This trait will live 
+in sawtooth-lib and is used to define one of the generics for the publisher.
 
 ```rust
-// This trait would go in sawtooth-lib
-pub trait Batch<T: Transaction>: Send {
-    fn id(&self) -> &str;
-
-    fn transactions(&self) -> &[T];
-}
-
-pub trait Transaction: Send {
-    fn id(&self) -> &str;
-
-    fn payload(&self) -> &[u8];
-
-    fn header(&self) -> &[u8];
-}
+pub trait Batch: Send {}
 ```
 
-#### PublishedResultCreator
+#### ArtifactCreator
 
-This trait will be Boxed and used to generate the resulting `PublishedResult`
+This trait will be Boxed and used to generate the resulting `Artifact`
 from the resulting execution results.
 
 ```rust
-
-pub trait PublishedResultCreatorFactory<
-    B: Batch<T>,
-    C: PublisherContext<B, T>,
-    R: PublishedResult,
-    T: Transaction,
->
-{
-    fn new_creator(&self) -> Result<Box<dyn PublishedResultCreator<B, C, R, T>>, InternalError>;
+/// An artifact creator.
+///
+/// Used to create new instances of `Artifact`.
+pub trait ArtifactCreator: Send {
+    /// The context that contains extraneous information required to create the
+    /// `Artifact`.
+    type Context;
+    /// The type of input for the specific `Artifact`
+    type Input;
+    /// The `Artifact` type
+    type Artifact;
+
+    /// Creates a new `Artifact`
+    ///
+    /// Returns a an `Artifact` created from the context and input
+    fn create(
+        &self,
+        context: &mut Self::Context,
+        input: Self::Input,
+    ) -> Result<Self::Artifact, InternalError>;
 }
-
 ```
 
 The publisher will receive a factory for creating this creator, as one is
 required per call to `start()`.
 
 ```rust
-pub trait PublishedResultCreator<
-    B: Batch<T>,
-    C: PublisherContext<B, T>,
-    R: PublishedResult,
-    T: Transaction,
->: Send
-{
-    fn create(
-        &self,
-        context: C,
-        batch_results: Vec<BatchExecutionResult<B, T>>,
-        resulting_state_root: String,
-    ) -> Result<R, InternalError>;
-}
+/// An artifact creator factory.
+///
+/// Used to create new instances of `ArtifactCreator`.
+pub trait ArtifactCreatorFactory {
+    /// The `ArtifactCreator` type
+    type ArtifactCreator;
+
+    /// Returns a new `ArtifactCreator`
+    fn new_creator(&self) -> Result<Self::ArtifactCreator, InternalError>;
 }
 ```
 
 #### PendingBatches
 This trait will be used to get the pending batches, from the `PendingQueue` for
-example. This is similar to an `Iterator` but needs to be `Send`. A Boxed
+example. This is similar to an `Iterator` but needs to be `Send`. A `Boxed`
```

Comment:
Not a blocker but `Boxed` maybe shouldn't be in backquotes?

Reply:
fixed